internal/sirius/get_firm_deputies.go

```diff
@@ -0,0 +1,133 @@
+package sirius
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+)
+
+type orderStatus struct {
+	Handle string `json:"handle"`
+	Label  string `json:"label"`
+}
+
+type client struct {
+	Id int `json:"id"`
+}
+
+type order struct {
+	Id          int         `json:"id"`
+	Client      client      `json:"client"`
+	OrderStatus orderStatus `json:"orderStatus"`
+}
+
+type orders struct {
+	Order order `json:"order"`
+}
+
+type executiveCaseManager struct {
+	EcmId   int    `json:"id"`
+	EcmName string `json:"displayName"`
+}
+
+type Deputies struct {
+	DeputyId             int                  `json:"id"`
+	Firstname            string               `json:"firstname"`
+	Surname              string               `json:"surname"`
+	DeputyNumber         int                  `json:"deputyNumber"`
+	Orders               []orders             `json:"orders"`
+	ExecutiveCaseManager executiveCaseManager `json:"executiveCaseManager"`
+	OrganisationName     string               `json:"organisationName"`
+}
+
+type FirmDeputy struct {
+	DeputyId             int
+	Firstname            string
+	Surname              string
+	DeputyNumber         int
+	ActiveClientsCount   int
+	ExecutiveCaseManager string
+	OrganisationName     string
+}
+
+func (c *Client) GetFirmDeputies(ctx Context, firmId int) ([]FirmDeputy, error) {
+	req, err := c.newRequest(ctx, http.MethodGet, fmt.Sprintf("/api/v1/firms/%d/deputies", firmId), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return nil, ErrUnauthorized
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, newStatusError(resp)
+	}
+
+	var v []Deputies
+	if err = json.NewDecoder(resp.Body).Decode(&v); err != nil {
+		return nil, err
+	}
+
+	var deputies []FirmDeputy
+
+	for _, t := range v {
+		var deputy = FirmDeputy{
+			DeputyId:             t.DeputyId,
+			Firstname:            t.Firstname,
+			Surname:              t.Surname,
+			DeputyNumber:         t.DeputyNumber,
+			ActiveClientsCount:   getActiveClientCount(t.Orders),
+			ExecutiveCaseManager: t.ExecutiveCaseManager.EcmName,
+			OrganisationName:     t.OrganisationName,
+		}
+
+		deputies = append(deputies, deputy)
+	}
+	sortedDeputies := sortTheDeputiesByNumberOfClients(deputies)
```

Comment:
Is the sorting of deputies by number of clients something that was added to the ticket retrospectively? I can't see a reference to it in the ticket or prototype so wondering why they are being sorted that way

Reply:
We did it because the ticket wants the deputies with 0 clients to be at the bottom of the list, so using a sort seemed like the simplest way to do it.